### PR TITLE
feat(FOROME-299): remove load and apply buttons from refiner control

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-299

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/utils/filter-refiner/compareConditions.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  currentPreset,
+  prevPreset,
+}: ICompareConditionsProps): boolean => {
+  if (currentConditions.length !== startConditions.length) return true
+
+  if (prevPreset && currentPreset !== prevPreset) return true
+
+  let acc = 0
+
+  currentConditions.forEach((condition: any[], index) => {
```
Comment (AndreyUstyumenko) [inline src/utils/filter-refiner/compareConditions.ts]: use reduce

File: src/store/filterPreset.ts [hunk 2]
```diff
@@ -96,6 +93,67 @@ class PresetStore {
...
+    this.updatePresetAsync(preset)
+
+    filterStore.resetActionName()
+
+    showToast(t('header.presetFilterAction.modify'), 'success')
+
+    preset && datasetStore.setPrevPreset(preset)
+  }
+
+  handleLoadPreset(preset: string): void {
```
Comment (AndreyUstyumenko) [inline src/store/filterPreset.ts]: naming issue: store should contain only actions, not handlers

File: src/pages/filter/ui/filter-control-refiner.tsx [hunk 3]
```diff
@@ -14,68 +15,71 @@ import { DropDown } from '@ui/dropdown'
...
-  const presets: string[] = get(datasetStore, 'dsStat.filter-list', [])
-    .filter((preset: FilterList) => {
-      if (
-        filterStore.actionName === ActionFilterEnum.Delete ||
-        filterStore.actionName === ActionFilterEnum.Modify
-      ) {
-        return !preset.standard
-      }
+  const isApplyDisabled =
+    activePreset.startsWith('⏚') &&
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/filter-control-refiner.tsx]: better to move ⏚ sign to a const

File: src/pages/filter/ui/filter-control-refiner.tsx [hunk 4]
```diff
@@ -14,68 +15,71 @@ import { DropDown } from '@ui/dropdown'
...
 export const FilterControlRefiner = observer((): ReactElement => {
-  const [activePreset, setActivePreset] = useState(datasetStore.activePreset)
-  const [createPresetName, setCreatePresetName] = useState('')
+  const activePreset: string = datasetStore.activePreset
+
+  const [createPresetName, setCreatePresetName] = useState<string>('')
+
+  const isSelectedFiltersEmpty: boolean = isEmpty(filterStore.selectedFilters)
+
+  const canBeModified = (preset: FilterList): boolean => {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/filter-control-refiner.tsx]: could you refactor this stuff and create a store? Seems like it's not much work

**Review by AndreyUstyumenko — APPROVED** [review]

File: src/utils/filter-refiner/compareConditions.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  currentConditions,
+  startConditions,
+  currentPreset,
+  prevPreset,
+}: ICompareConditionsProps): boolean => {
+  if (currentConditions.length !== startConditions.length) return true
+
+  if (prevPreset && currentPreset !== prevPreset) return true
+
+  const sum = currentConditions.reduce(
```
Comment (AlMaQntr) [inline src/utils/filter-refiner/compareConditions.ts]: In this code you count the number of elements that satisfy the condition `(difference(condition[index], startConditions[index]).length === 0)`
And than compares this number with array length.
May be we can use `every` method of array?

File: src/utils/filter-refiner/compareConditions.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        acc += 1
+      }
+      return acc
+    },
+    0,
+  )
+
+  if (sum === currentConditions.length) return false
+
+  return false
```
Comment (AlMaQntr) [inline src/utils/filter-refiner/compareConditions.ts]: Very strange `return false` after conditional `return false`. Just return false without conditions? Or is it an algorithmic mistake?

**Review by AlMaQntr — APPROVED** [review]

**Pull request merged**